### DISCUSSION TITLE
Recommended or best practices for deploying module updates

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#14648] [General]
Are there are general guidelines or recommendations on how to deploy updates for modules?

**Comment by Piedone** [comment]
You'd keep a module as part of a larger ASP.NET Core app that uses Orchard Core. So if by "deploy" you mean deploying a new version of your app, together with updates to your modules, to a server, then it's just ASP.NET Core publishing.

Or do you mean what should happen when you change the DB schema or content definition corresponding to a module? You need migrations for that, see e.g. https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Migrations/BookMigrations.cs and https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Migrations/PersonMigrations.cs.

**Reply by weirdyang** [reply]
thanks, I was hoping for a way to deploy changes without republishing the web project, for example by just dropping in the updated dlls.

**Reply by Piedone** [reply]
You can do that, but it's really risky. I recommend you do a proper publishing.

**Reply by weirdyang** [reply]
Thanks!
